src/webgpu/api/operation/vertex_state.spec.ts

```diff
@@ -0,0 +1,348 @@
+
+    this.colorAttachment = this.device.createTexture({
+      format: kTextureFormat,
+      size: { width: kWidth, height: kHeight, depth: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+    });
+
+    this.result = this.device.createBuffer({
+      size: this.byteLength,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    });
+  }
+
+  MakeRenderPipeline(
+    primitiveTopology: GPUPrimitiveTopology,
+    indexFormat?: GPUIndexFormat
+  ): GPURenderPipeline {
+    const vertexModule = this.device.createShaderModule({
+      code: `
+        [[location(0)]] var<in> pos : vec4<f32>;
+        [[builtin(position)]] var<out> Position : vec4<f32>;
+
+        [[stage(vertex)]]
+        fn main() -> void {
+          Position = pos;
+          return;
+        }
+      `,
+    });
+
+    const fragmentModule = this.device.createShaderModule({
+      code: `
+        [[location(0)]] var<out> fragColor : vec4<f32>;
+
+        [[stage(fragment)]]
+        fn main() -> void {
+          fragColor = vec4<f32>(0.0, 1.0, 0.0, 1.0);
+          return;
+        }
+      `,
+    });
+
+    return this.device.createRenderPipeline({
+      layout: this.device.createPipelineLayout({ bindGroupLayouts: [] }),
+      vertexStage: { module: vertexModule, entryPoint: 'main' },
+      fragmentStage: { module: fragmentModule, entryPoint: 'main' },
+      primitiveTopology,
+      colorStates: [{ format: kTextureFormat }],
+      vertexState: {
+        indexFormat,
+        vertexBuffers: [
+          {
+            arrayStride: 4 * 4,
+            stepMode: 'vertex',
+            attributes: [{ format: 'float4', offset: 0, shaderLocation: 0 }],
+          },
+        ],
+      },
+    });
+  }
+
+  CreateIndexBuffer(indexArray: Uint16Array | Uint32Array, indexFormat: GPUIndexFormat): GPUBuffer {
+    const indexBuffer = this.device.createBuffer({
+      size: indexArray.byteLength,
+      usage: GPUBufferUsage.INDEX,
+      mappedAtCreation: true,
+    });
+    if (indexFormat === 'uint16') {
+      new Uint16Array(indexBuffer.getMappedRange()).set(indexArray);
+    } else {
+      new Uint32Array(indexBuffer.getMappedRange()).set(indexArray);
+    }
+    indexBuffer.unmap();
+    return indexBuffer;
+  }
+
+  run(
+    indexBuffer: GPUBuffer,
+    indexCount: number,
+    indexFormat: GPUIndexFormat,
+    indexOffset: number = 0,
+    primitiveTopology: GPUPrimitiveTopology = 'triangle-list'
+  ): void {
+    let pipeline: GPURenderPipeline;
+    // The indexFormat must be set in render pipeline descriptor that specifys a strip primitive
+    // topology for primitive restart testing
+    if (primitiveTopology === 'line-strip' || primitiveTopology === 'triangle-strip') {
+      pipeline = this.MakeRenderPipeline(primitiveTopology, indexFormat);
+    } else {
+      pipeline = this.MakeRenderPipeline(primitiveTopology);
+    }
+
+    /* prettier-ignore */
+    const vertexArray = new Float32Array([
```

Comment:
uNit:
```suggestion
    const vertexArray = new Float32Array(/* prettier-ignore */ [
```
so it applies only to the array `[...]` instead of the whole `const ...;`